src/app/toy/controllers/choose-favourite.js

```diff
@@ -0,0 +1,23 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+
+class ConfirmController extends BaseController {
+  async saveValues(req, res, callback) {
+    super.saveValues(req, res, async () => {
+      const headers = sessionId
+        ? { "session-id": req.session.tokenId }
+        : undefined; // set the header to null should fail the req but pass the bro
```

Comment:
Am not a fan of setting javascript things to `undefined` (i know we've been doing this even in typescript) i'll rather set it to `null` instead to show it's a deliberate decision by the developer and reserve undefined for the javascript engine

Reply:
Interesting. I almost never use `null` as there can be weird issues using it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null#difference_between_null_and_undefined

```
typeof null; // "object" (not "null" for legacy reasons)
typeof undefined; // "undefined"
```

These two differences have cause me issues before.

https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Obviously, the better approach is to move this to TypeScript and get intentional & planned out typing, not the language features that got rushed out the door in 1995.

Reply:
Yeah, I forgot all about the null issues.